### ISSUE TITLE
Commas in amount are not handled well

### ISSUE THREAD

**Issue by judgej** [thephpleague/omnipay-common#17]
Making a payment for "1,234.56" will actually submit a payment to the gateway for "1.00".

The issue is caused by PHP's number conversion, meaning that the string conversion breaks at the first comma, and the remainder is not checked.

Now, validation and number checking and conversion should be done before the amount is passed OmniPay BUT OmniPay accepts a string, and how it handles that string should be a bit more predictable. "1,234.56" should either be sent to the gateway as "1234.56" or an exception should be raised. Sending "1.00" is just a surprise waiting to bite someone. Yes, it bit us.

**Comment by alauddinmondal** [comment]
I am adding my amount from html form data as $amount = $_POST['amount'].'0.'; I have entered AED 100,
but my problem is here the amount is showing 1000.00 this is not showing that I have entered in html form

**Comment by judgej** [comment]
@alauddinmondal Not sure why you have posted here rather than raising a new issue, but you need to check a little further. The first thing you need to find out, is *exactly* what is in `$amount`. That way you will know which side of this line of code the problem is in - your form, or OmniPay.

**Comment by alauddinmondal** [comment]
Thank you sir this is working fine. Everything is very nice

**Comment by alauddinmondal** [comment]
thank a lot

**Comment by delatbabel** [comment]
> $amount = $_POST['amount'].'0.';

Please see https://xkcd.com/327/

**Comment by alauddinmondal** [comment]
Very very nice we are the GitHub

**Comment by judgej** [comment]
@alauddinmondal In case the xkcd link was not clear, don't use `$_POST` without some kind of validation of what is being passed in first, *anywhere* in your code. So in this case if you are expecting only an integer, than make sure it is an integer only.
